src/icrc2-minter/src/tokens/icrc1.rs

```diff
@@ -104,20 +113,99 @@ async fn query_icrc1_configuration(token: Principal) -> Result<TokenConfiguratio
 }
 
 /// Requests fee and minting account configuration from an ICRC-1 canister.
-async fn query_icrc1_token_info(token: Principal) -> Result<TokenInfo> {
-    let icrc_client = IcrcCanisterClient::new(IcCanisterClient::new(token));
+///
+/// `has_token_info_in_metadata` is a flag that indicates whether the token info is set in metadata.
+/// If it is, we can use the standard ICRC-1 metadata query to get the token info.
+/// Otherwise, we have to use dedicated queries to get the token info.
+async fn query_icrc1_token_info<C>(
+    icrc_client: &IcrcCanisterClient<C>,
+    has_token_info_in_metadata: bool,
+) -> Result<TokenInfo>
+where
+    C: CanisterClient,
+{
+    // If the token info is set in metadata, we can use the standard ICRC-1 metadata query.
```

Comment:
Can we check which tokens don't have the info set in the metadata, this should be a requirement for all tokens.  In that case, is it necssary to have: query_icrc1_token_info_from_dedicated_queries ?

Reply:
In general I haven't found any ICRC which doesn't expose those keys in the metadata, but the ICRC standard states:

```md
ledger can expose metadata to simplify integration with wallets and improve user experience. The client can use the [icrc1_metadata](https://github.com/dfinity/ICRC-1/tree/main/standards/ICRC-1#metadata_method) method to fetch the metadata entries. All the metadata entries are optional.
```

So they are actually optional, even if in almost all cases they are implemented

Reply:
I think if metadata is not set, we shouldn't allow bridging.

Reply:
Imo we should stick to the ICRC standard, which suggests but doesn't force to include them. Anyway probably they are set in 99% of the icrc token.